common/app/rendering/core/JavascriptProps.scala

```diff
@@ -0,0 +1,22 @@
+package rendering.core
+
+import conf.Configuration
+import play.api.libs.json.{Json, Writes}
+
+case class JavascriptPropsConfig(
+  bundleUrl: String = conf.Static("javascripts/ui.bundle.browser.js"),
+  polyfillioUrl: String =
+    if (conf.switches.Switches.PolyfillIO.isSwitchedOn) common.Assets.js.polyfillioUrl
+    else conf.Static("javascripts/vendor/polyfillio.fallback.js"),
+  beaconUrl: String = Configuration.debug.beaconUrl
+)
+case class JavascriptProps(
+  config: JavascriptPropsConfig = JavascriptPropsConfig()
+)
+
+object JavascriptPropsConfig {
+  implicit val javascriptPropsConfig: Writes[JavascriptPropsConfig] = Json.writes[JavascriptPropsConfig]
+}
+object JavascriptProps {
```

Comment:
You don't need separate `object` you can probably put the 2 implicits in a single `JavascriptProps` object or something similar